pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -85,23 +83,12 @@ func (r *ReconcileCheInstallation) Reconcile(request reconcile.Request) (reconci
 		return reconcile.Result{}, err
 	}
 
-	var errors []error
 	if isResourceCreated, err := r.EnsureCheSubscription(reqLogger, cheInstallation); err != nil {
-		errors = append(errors, err)
-	} else if isResourceCreated {
-		return reconcile.Result{Requeue: true}, nil
-	}
-
-	if isResourceCreated, err := r.EnsureTektonSubscription(reqLogger, cheInstallation); err != nil {
-		errors = append(errors, err)
+		return reconcile.Result{}, err
 	} else if isResourceCreated {
 		return reconcile.Result{Requeue: true}, nil
```

Comment:
The main reason for doing it differently here was the fact that we need to create the namespace only once when we installing the operator. Having a permanent watcher may be an overkill in this case. In member operator it's different. It's not one time action there.
On the other hand if the namespace get deleted then we won't restore it automatically without a watcher.
Anyway, I think it's OK if we keep it as is for now in this PR because it's about moving existing logic to a different controller. And we can modify the logic itself in a separate PR.

Reply:
Yes, This is what my plan is to have PR small